docs/content/docs/dev/table/sql/alter.md

```diff
@@ -125,38 +179,310 @@ tables = table_env.list_tables()
+|     Orders |
++------------+
+1 row in set
 
 Flink SQL> ALTER TABLE Orders RENAME TO NewOrders;
-[INFO] Table has been removed.
+[INFO] Execute statement succeed.
 
 Flink SQL> SHOW TABLES;
-NewOrders
++------------+
+| table name |
++------------+
+|  NewOrders |
++------------+
+1 row in set
 ```
 {{< /tab >}}
 {{< /tabs >}}
 
+{{< top >}}
+
 ## ALTER TABLE
 
-* Rename Table
+The following grammar gives an overview about the available syntax:
+```text
+ALTER TABLE [IF EXISTS] table_name {
+    ADD { <schema_component> | (<schema_component> [, ...]) }
+  | MODIFY { <schema_component> | (<schema_component> [, ...]) }
+  | DROP {column_name | (column_name, column_name, ....) | PRIMARY KEY | CONSTRAINT constraint_name | WATERMARK}
+  | RENAME old_column_name TO new_column_name
+  | RENAME TO new_table_name
+  | SET (key1=val1, ...)
+  | RESET (key1, ...)
+}
 
-```sql
-ALTER TABLE [catalog_name.][db_name.]table_name RENAME TO new_table_name
+<schema_component>:
+  { <column_component> | <constraint_component> | <watermark_component> }
+
+<column_component>:
+  column_name <column_definition> [FIRST | AFTER column_name]
+
+<constraint_component>:
+  [CONSTRAINT constraint_name] PRIMARY KEY (column_name, ...) NOT ENFORCED
+
+<watermark_component>:
+  WATERMARK FOR rowtime_column_name AS watermark_strategy_expression
+
+<column_definition>:
+  { <physical_column_definition> | <metadata_column_definition> | <computed_column_definition> } [COMMENT column_comment]
+
+<physical_column_definition>:
+  column_type
+
+<metadata_column_definition>:
+  column_type METADATA [ FROM metadata_key ] [ VIRTUAL ]
+
+<computed_column_definition>:
+  AS computed_column_expression
 ```
 
-Rename the given table name to another new table name.
+<span class="label label-info">Note</span> If the table does not exist, `ValidationException` is thrown unless `IF EXISTS` is specified.
 
-* Set or Alter Table Properties
+**Add Table Column**
 
-```sql
-ALTER TABLE [catalog_name.][db_name.]table_name SET (key1=val1, key2=val2, ...)
+```text
+ALTER TABLE [IF EXISTS] [catalog_name.][db_name.]table_name ADD (<column_component>[, <column_component>, ...])
+```
+
+Add table column(s) to the specified position, which includes
+- Append the column to the last position
+- Insert the column to the first position
+- Insert the column after the specified column
+
+Throws `ValidationException` when either of following conditions are met
+- Add a column name which is already defined in the table schema
+- Add a column which refers to a nonexistent column by `AFTER`
+- Add a computed column which is derived from another computed or nonexistent columns
+- Add a computed column with invalid expression
+
+**Add Table Constraint**
+
+```text
+ALTER TABLE [IF EXISTS] [catalog_name.][db_name.]table_name ADD [CONSTRAINT constraint_name] PRIMARY KEY(col1[, col2, ...]) NOT ENFORCED
+```
+
+Add primary key to the table. Throws `ValidationException` when either of following conditions are met
```

Comment:
It's better if we can refer to the create#PRIMARY KEY section. I think we have the same rules when creating a table with pk.